src/core/sparsemat.c

```diff
@@ -700,11 +700,18 @@ int igraph_sparsemat_fkeep(igraph_sparsemat_t *A,
                            int (*fkeep)(int, int, igraph_real_t, void*),
                            void *other) {
 
-    if (!cs_fkeep(A->cs, fkeep, other)) {
-        IGRAPH_ERROR("Cannot filter sparse matrix", IGRAPH_FAILURE);
+    if (!A) {
+        IGRAPH_ERROR("The sparse matrix pointer is NULL.", IGRAPH_EINVAL);
     }
+    if (A->cs->nz != -1) {
+        IGRAPH_ERROR("The sparse matrix is not in compressed format.", IGRAPH_EINVAL);
```

Comment:
I would use `nz >= 0` instead of `n != -1`, as this is what the CS library itself does internally. I don't know how much we can trust the claim from the comment that this is always precisely -1 (and not say -2).

https://github.com/igraph/igraph/blob/master/vendor/cs/cs.h#L658

Reply:
Actually, I was wrong. They use `== -1` just one line above.

https://github.com/igraph/igraph/blob/master/vendor/cs/cs.h#L657